Patch originally from Gentoo:
=============================
Fix buffer overflow with GCC 4.5 -D_FORTIFY_SOURCE=2.
https://bugs.gentoo.org/317727
https://sourceforge.net/tracker/?func=detail&aid=3048354&group_id=10894&atid=310894

diff -p -up ./generic/tclTrace.c.fortify ./generic/tclTrace.c
--- ./generic/tclTrace.c.fortify	2008-10-26 19:34:04.000000000 +0100
+++ ./generic/tclTrace.c	2010-12-06 18:42:22.000000000 +0100
@@ -908,7 +908,7 @@ TraceVariableObjCmd(
 	    }
 	    ctvarPtr->traceCmdInfo.length = length;
 	    flags |= TCL_TRACE_UNSETS | TCL_TRACE_RESULT_OBJECT;
-	    strcpy(ctvarPtr->traceCmdInfo.command, command);
+	    memcpy(ctvarPtr->traceCmdInfo.command, command, length + 1);
 	    ctvarPtr->traceInfo.traceProc = TraceVarProc;
 	    ctvarPtr->traceInfo.clientData = &ctvarPtr->traceCmdInfo;
 	    ctvarPtr->traceInfo.flags = flags;

